### ISSUE TITLE
OCP 4.10 Release Notes Tracker

### ISSUE THREAD

**Issue by stevsmit** [openshift/openshift-docs#37586]
Please leave comments here for anything that should be highlighted in the 4.10 release notes. If possible, provide a link to the Jira or BZ related to your item. Thank you!

**A note about migrating to Jira**
This tracker might be moved to Jira. In the event that happens, I will update accordingly and provide a direct link. 

**A note about known issues**
All Bugzilla items that need to be documented as known issues in the release notes should have Doc Type = Known Issue and the Doc Text field populated accordingly. Before adding a known issue to this tracker:

1. Ensure the Doc Type and Doc Text fields are filled in correctly.
2. See if your issue shows up in the query we will use to populate the release notes.
3. If the fields are set correctly, and the issue is not in the query, add it here.

**Resources**

- [Feature GA Status Tracker](https://docs.google.com/spreadsheets/d/1N1JJi7roK0BUFYo-PKI6wT3EMG5XD1j3eFNNEhmVnJc/edit#gid=0)

**Comment by atiratree** [comment]
added support for podman config credentials in `oc`
- `oc` commands that use registry config (`oc login`, `oc image`, etc.) now obtain credentials from podman config locations if a registry entry cannot be found in the default docker config location.
- The order can be changed via `REGISTRY_AUTH_PREFERENCE` env variable to docker (current default - deprecated) or podman (prioritizes podman credentials over docker).
- `REGISTRY_AUTH_FILE` env variable can be used instead of `--registry-config` option 


Deprecated features
docker config location deprecated in `oc`
- `oc` commands that use registry config currently obtain credentials from docker config location (usually `~/.docker/config.json`) by default. This is deprecated and will be removed in 2 releases in favor of podman config locations.


PR: https://github.com/openshift/oc/pull/893 and https://github.com/openshift/oc/pull/983

**Comment by soltysh** [comment]
oc 4.10 is the last release being built with support for rhel7, starting from 4.11 we will provide only rhel8 compatible builds. This is coming from the fact that oc is statically linking glibc. More information can be found in https://developers.redhat.com/blog/2020/10/08/migrating-c-and-c-applications-from-red-hat-enterprise-linux-version-7-to-version-8#

**Comment by mtulio** [comment]
installer changed the default instance type on AWS nodes to m6i: m6i.xlarge for control-plane and m6.large for compute nodes.
- PR: https://github.com/openshift/installer/pull/5327
- Card: https://issues.redhat.com/browse/SPLAT-329

installer changed the default block storage on AWS nodes from gp2 to gp3
- PR: https://github.com/openshift/installer/pull/5239
- Cards: [SPLAT-253](https://issues.redhat.com/browse/SPLAT-253) [SPLAT-343](https://issues.redhat.com/browse/SPLAT-343)

**Comment by ptrnull** [comment]
The release notes indicate that OpenShift 4.10 is released based on Kubernetes 1.22 but the FC release is using Kubernetes 1.23: https://openshift-release-artifacts.apps.ci.l2s4.p1.openshiftapps.com/4.10.0-fc.1/release.txt

**Comment by s-urbaniak** [comment]
Starting from OpenShift 4.10 control plane components have no fallback mechanisms included any more to fall back to the Common Name field on X.509 certificates when no Subject Alternative Names (SANs) are present. Any certificates must properly set the Subject Alternative Names field. This was initially announced in OpenShift 4.6, see https://docs.openshift.com/container-platform/4.6/release_notes/ocp-4-6-release-notes.html#ocp-4-6-tls-common-name.

**Comment by SNiemann15** [comment]
Working together with @ktania46 and @alishaIBM to add IBM Z and Power content to the RNs: 
https://docs.google.com/document/d/1LZqUJ86zmUt2Erh1mjhrACEbfBHEoE4z2Hgnr_MrugU/edit#heading=h.gi343gic2seb

**Comment by jechen0648** [comment]
EgressIP on plubic cloud: https://issues.redhat.com/browse/SDN-1316 https://issues.redhat.com/browse/SDN-1315, https://issues.redhat.com/browse/SDN-1332,

**Comment by duanwei33** [comment]
Strorge part:
CSI Azure Disk Driver Operator  GA in 4.10  https://issues.redhat.com/browse/STOR-552
CSI OpenStack Cinder Driver Operator  GA from 4.8  https://issues.redhat.com/browse/STOR-679
CSI AWS EFS Driver Operator  GA in 4.10  https://issues.redhat.com/browse/STOR-362
CSI vSphere Driver Operator  GA in 4.10  https://issues.redhat.com/browse/STOR-558  
CSI Alicloud Disk operator  GA in 4.10  https://issues.redhat.com/browse/STOR-574
CSI IBM Cloud Operator  GA in 4.10  https://issues.redhat.com/browse/STOR-598
CSI Azure File Operator  TP in 4.10  https://issues.redhat.com/browse/STOR-555
Automatic device discovery and provisioning with Local Storage Operator  TP in 4.10

**Comment by rioliu-rh** [comment]
for config drift part, if [pr#2910](https://github.com/openshift/machine-config-operator/pull/2910) cannot be merged before code freeze date Jan 28, we need to add [bz2038240](https://bugzilla.redhat.com/show_bug.cgi?id=2038240) to release notes as known issue

**Comment by quarterpin** [comment]
The ingress now adds support for IBM cloud environment: https://issues.redhat.com/browse/NE-619
The telemetry now includes metrics to track the number of active NLB routers in an AWS environment: https://issues.redhat.com/browse/NE-704

**Comment by xingxingxia** [comment]
1 PodSecurity is temporarily disabled in 4.10 and will be enabled in 4.11, see bug [2008462](https://bugzilla.redhat.com/show_bug.cgi?id=2008462) . It is [promoted to beta in 1.23](https://kubernetes.io/docs/concepts/security/pod-security-admission/#enabling-the-podsecurity-admission-plugin ), and 4.10 is k8s 1.23, so we can mention this (Discussed with Dev in Slack).
2 Add notes for OpenID IDP group consumption [AUTH-8](https://issues.redhat.com/browse/AUTH-8)

**Comment by rioliu-rh** [comment]
> for config drift part, if [pr#2910](https://github.com/openshift/machine-config-operator/pull/2910) cannot be merged before code freeze date Jan 28, we need to add [bz2038240](https://bugzilla.redhat.com/show_bug.cgi?id=2038240) to release notes as known issue

PR is merged and BZ is verified. don't need to add it to release notes

**Comment by gpei** [comment]
From Installation 

- Allow thin provisioning for VMware vSphere IPI - https://issues.redhat.com/browse/CORS-1674
- Azure: IPI Support for Azure Stack Hub - https://issues.redhat.com/browse/CORS-1596
- IBM Public Cloud: Support IPI deployment - https://issues.redhat.com/browse/CORS-1513
- Support Alibaba IPI installer - https://issues.redhat.com/browse/SPLAT-193

**Comment by bgilbert** [comment]
`coreos-installer` now has `iso customize` and `pxe customize` commands for flexible customization of bare metal RHCOS installation from the live ISO/PXE images ([COS-823](https://issues.redhat.com/browse/COS-823)).

**Comment by quarterpin** [comment]
List of features that are to be introduced with v4.10 for Network Edge component: 
https://issues.redhat.com/browse/NE-542 - support to have ROUTER_ENABLE_COMPRESSION and ROUTER_COMPRESSION_MIME options in HAProxy
https://issues.redhat.com/browse/NE-624 - Support a Configurable syslog Maximum Message Size for HAProxy
https://issues.redhat.com/browse/NE-587 - Increase the Length of HAproxy Request URI in Logs

https://issues.redhat.com/browse/NE-505 - OpenShift Network Edge Support for Alibaba Cloud Platform
https://issues.redhat.com/browse/NE-621 - Support mutating .spec.endpointPublishingStrategy
https://issues.redhat.com/browse/NE-303 - Tech Preview introduction of ExternalDNS
https://issues.redhat.com/browse/NE-679 - Set CoreDNS forwarding policy via the DNS operator

https://issues.redhat.com/browse/NE-684 -  Add support to update upstream resolvers in the pod [coreDNS]
https://issues.redhat.com/browse/NE-367 - [Observability] Add logLevel and operatorLogLevel APIs for DNS

**Comment by cowgirljessi** [comment]
> List of features that are to be introduced with v4.10 for Network Edge component: https://issues.redhat.com/browse/NE-542 - support to have ROUTER_ENABLE_COMPRESSION and ROUTER_COMPRESSION_MIME options in HAProxy https://issues.redhat.com/browse/NE-624 - Support a Configurable syslog Maximum Message Size for HAProxy https://issues.redhat.com/browse/NE-587 - Increase the Length of HAproxy Request URI in Logs
> 
> https://issues.redhat.com/browse/NE-505 - OpenShift Network Edge Support for Alibaba Cloud Platform https://issues.redhat.com/browse/NE-621 - Support mutating .spec.endpointPublishingStrategy https://issues.redhat.com/browse/NE-303 - Tech Preview introduction of ExternalDNS https://issues.redhat.com/browse/NE-679 - Set CoreDNS forwarding policy via the DNS operator
> 
> https://issues.redhat.com/browse/NE-684 - Add support to update upstream resolvers in the pod [coreDNS] https://issues.redhat.com/browse/NE-367 - [Observability] Add logLevel and operatorLogLevel APIs for DNS

I'm working on the NE-367 documentation.

**Comment by xenolinux** [comment]
> List of features that are to be introduced with v4.10 for Network Edge component: https://issues.redhat.com/browse/NE-542 - support to have ROUTER_ENABLE_COMPRESSION and ROUTER_COMPRESSION_MIME options in HAProxy https://issues.redhat.com/browse/NE-624 - Support a Configurable syslog Maximum Message Size for HAProxy https://issues.redhat.com/browse/NE-587 - Increase the Length of HAproxy Request URI in Logs
> 
> https://issues.redhat.com/browse/NE-505 - OpenShift Network Edge Support for Alibaba Cloud Platform https://issues.redhat.com/browse/NE-621 - Support mutating .spec.endpointPublishingStrategy https://issues.redhat.com/browse/NE-303 - Tech Preview introduction of ExternalDNS https://issues.redhat.com/browse/NE-679 - Set CoreDNS forwarding policy via the DNS operator
> 
> https://issues.redhat.com/browse/NE-684 - Add support to update upstream resolvers in the pod [coreDNS] https://issues.redhat.com/browse/NE-367 - [Observability] Add logLevel and operatorLogLevel APIs for DNS

I am working on https://issues.redhat.com/browse/NE-753

**Comment by skrthomas** [comment]
> List of features that are to be introduced with v4.10 for Network Edge component: https://issues.redhat.com/browse/NE-542 - support to have ROUTER_ENABLE_COMPRESSION and ROUTER_COMPRESSION_MIME options in HAProxy https://issues.redhat.com/browse/NE-624 - Support a Configurable syslog Maximum Message Size for HAProxy https://issues.redhat.com/browse/NE-587 - Increase the Length of HAproxy Request URI in Logs
> 
> https://issues.redhat.com/browse/NE-505 - OpenShift Network Edge Support for Alibaba Cloud Platform https://issues.redhat.com/browse/NE-621 - Support mutating .spec.endpointPublishingStrategy https://issues.redhat.com/browse/NE-303 - Tech Preview introduction of ExternalDNS https://issues.redhat.com/browse/NE-679 - Set CoreDNS forwarding policy via the DNS operator
> 
> https://issues.redhat.com/browse/NE-684 - Add support to update upstream resolvers in the pod [coreDNS] https://issues.redhat.com/browse/NE-367 - [Observability] Add logLevel and operatorLogLevel APIs for DNS

There are no Network Edge docs for https://issues.redhat.com/browse/NE-505 - OpenShift Network Edge Support for Alibaba Cloud Platform for 4.10.

**Comment by bmcelvee** [comment]
> List of features that are to be introduced with v4.10 for Network Edge component: https://issues.redhat.com/browse/NE-542 - support to have ROUTER_ENABLE_COMPRESSION and ROUTER_COMPRESSION_MIME options in HAProxy https://issues.redhat.com/browse/NE-624 - Support a Configurable syslog Maximum Message Size for HAProxy https://issues.redhat.com/browse/NE-587 - Increase the Length of HAproxy Request URI in Logs
> 
> https://issues.redhat.com/browse/NE-505 - OpenShift Network Edge Support for Alibaba Cloud Platform https://issues.redhat.com/browse/NE-621 - Support mutating .spec.endpointPublishingStrategy https://issues.redhat.com/browse/NE-303 - Tech Preview introduction of ExternalDNS https://issues.redhat.com/browse/NE-679 - Set CoreDNS forwarding policy via the DNS operator
> 
> https://issues.redhat.com/browse/NE-684 - Add support to update upstream resolvers in the pod [coreDNS] https://issues.redhat.com/browse/NE-367 - [Observability] Add logLevel and operatorLogLevel APIs for DNS

The RN [PR](https://github.com/openshift/openshift-docs/pull/41323) is in progress for: https://issues.redhat.com/browse/NE-621 - Support mutating .spec.endpointPublishingStrategy

**Comment by ctauchen** [comment]
> EgressIP on plubic cloud: https://issues.redhat.com/browse/SDN-1316 https://issues.redhat.com/browse/SDN-1315, https://issues.redhat.com/browse/SDN-1332,

Documented in https://issues.redhat.com/browse/OSDOCS-2682
RN PR: https://github.com/openshift/openshift-docs/pull/39805

**Comment by ctauchen** [comment]
> for config drift part, if [pr#2910](https://github.com/openshift/machine-config-operator/pull/2910) cannot be merged before code freeze date Jan 28, we need to add [bz2038240](https://bugzilla.redhat.com/show_bug.cgi?id=2038240) to release notes as known issue

@rioliu-rh [pr#2910](https://github.com/openshift/machine-config-operator/pull/2910) merged 10 days ago, so it looks like we **don't** need to add [bz2038240](https://bugzilla.redhat.com/show_bug.cgi?id=2038240) as known issue.

Is that right? Thanks!

**Comment by kasturinarra** [comment]
Scheduler policy is removed in 4.10 we could add that to the table. @ingvagabund could you please help confirm ? Thanks !!

**Comment by ingvagabund** [comment]
Yes, the scheduler Policy API was removed in 4.10.

**Comment by kasturinarra** [comment]
I think we should mention that "Simplified disconnected mirroring with oc" is going to be TP release in 4.10. @afflom @zhouying7780 could you please help confirm ? thanks !!

**Comment by stevsmit** [comment]
@afflom @zhouying7780 please confirm on https://github.com/openshift/openshift-docs/pull/41590 if this is accurate. Thank you. cc @kasturinarra

**Comment by paigerube14** [comment]
Is there a plan to add information on how to create [machinesets](https://docs.openshift.com/container-platform/4.10/machine_management/creating_machinesets/creating-machineset-aws.html) and [infrastructure machine sets](https://docs.openshift.com/container-platform/4.10/machine_management/creating-infrastructure-machinesets.html) for IBM cloud since it's in the list for installation types?

**Comment by duanwei33** [comment]
For storage part during the upgrade:
what we have now:
```
==== CSI driver installation on vSphere clusters

To install a CSI driver on a cluster running on vSphere, you must have the following components installed:

* Virtual hardware version 15 or greater
* vSphere version 6.7 Update 3 or greater
* VMware ESXi version 6.7 Update 3 or greater

These components are still supported, but are deprecated; support for them will be removed in a future version of {product-title}.
```
After that, we need to add a note if the components are lower, upgrade from 4.9 to 4.10 on vSphere is supported, but the vSphere CSI Driver will not be installed on 4.10, also the cluster will be marked as unupgradeable.  
(due to https://bugzilla.redhat.com/show_bug.cgi?id=2048655)

**Comment by jotak** [comment]
Hey, netobserv engineering team here,
Not sure if this is eligible for the release note, our team has 2 enhancements delivered in the console for 4.10 (we previously notified the console team about them, cc @opayne1, if anywhere I think that should go under the Web console section ). Maybe they are not considered relevant enough?
They are:
- Show pods affected by a network policy in creation form (https://issues.redhat.com/browse/NETOBSERV-15)
- Switch between form and yaml views in network policy creation (https://issues.redhat.com/browse/NETOBSERV-6)

**Comment by afflom** [comment]
> I think we should mention that "Simplified disconnected mirroring with oc" is going to be TP release in 4.10. @afflom @zhouying7780 could you please help confirm ? thanks !!

That is correct. It is an oc plugin called oc-mirror.

**Comment by xiuwang** [comment]
> * `oc` commands that use registry config (`oc login`, `oc image`, etc.) now obtain credentials from podman config locations if a registry entry cannot be found in the default docker config location.

s/oc login/oc registry login
CC @atiratree Please help confirm, thanks!

**Comment by atiratree** [comment]
> s/oc login/oc registry login
> CC @atiratree Please help confirm, thanks!

@xiuwang Yup thanks for noticing that! Also two additional details:
- `oc registry login` : saves the credentials and not uses them. I am not sure how much that is important
- `oc image ...` is a family of commands and not just one command (this change is valid for all of them)

**Comment by bgilbert** [comment]
RHCOS PXE installation can now be configured to fetch from HTTPS servers that use a custom certificate authority or self-signed certificate.  This can be done by generating a customized initramfs image with `coreos-installer pxe customize --ignition-ca`.

(This is technically covered by https://github.com/openshift/openshift-docs/issues/37586#issuecomment-1024996962, but may be worth mentioning separately, since we've had a number of requests for it in [RFE-1098](https://issues.redhat.com/browse/RFE-1098).)

**Comment by stevsmit** [comment]
BZ and Known Issue at https://bugzilla.redhat.com/show_bug.cgi?id=2047732. cc @mjpytlak

**Comment by adambkaplan** [comment]
Storage:

Tech Preview feature - Shared Resource CSI Driver. This allows Secrets and ConfigMaps to be securely shared across namespaces. https://issues.redhat.com/browse/BUILD-293

**Comment by adambkaplan** [comment]
Builds:

Tech Preview - Builds can mount `csi` volumes. This can be used in conjunction with the Shared Resource CSI driver to use the cluster's Simple Content Access certificate within an OpenShift build. https://issues.redhat.com/browse/BUILD-274

**Comment by adambkaplan** [comment]
Jenkins (?):

The Jenkins Agent can be run as a sidecar container, allowing any container image to be run in a Jenkins pipeline with a correctly configured pod template and Jenkinsfile. https://issues.redhat.com/browse/JKNS-132

**Comment by chao007** [comment]
Storage:
IBM Cloud CSI Driver Operator should update to IBM VPC Block CSI Driver Operator

**Comment by ctauchen** [comment]
> IBM Cloud CSI Driver Operator

@chao007 I'm having trouble finding information about this. Anything you can share to point me in the right direction? Thanks!

**Comment by chao007** [comment]
> > IBM Cloud CSI Driver Operator
> 
> @chao007 I'm having trouble finding information about this. Anything you can share to point me in the right direction? Thanks!

@ctauchen We only discuss it in here [https://github.com/openshift/openshift-docs/pull/39815#discussion_r774267342]
```
oc get pods -n openshift-cluster-csi-drivers | grep operator
ibm-vpc-block-csi-driver-operator-7b9594cb4c-mdk76   1/1     Running   0          20h
```

**Comment by zhaozhanqi** [comment]
https://issues.redhat.com/browse/SDN-1817 - Ability to modify the Cluster Network MTU Post Installation 
this feature should mentioned in release note.  cc @brahaney

**Comment by yingwang-0320** [comment]
https://issues.redhat.com/browse/SDN-1353 - OVS kernel TC Flower HW Offload with CX-5: GA
Please add this feature in release note.

**Comment by anuragthehatter** [comment]
It might have covered as a sub by above various Alibaba RNs but worth to mentionOV support for Alibaba Platform

**Comment by yunjiang29** [comment]
@stevsmit [Bug 2051334](https://bugzilla.redhat.com/show_bug.cgi?id=2051334) - [AWS GovCloud] Cluster can not be installed on AWS GovCloud regions via terminal interactive UI. will not be fixed in 4.10 GA, please add it to the release note. thanks

sample description: Installing cluster into GovCloud regions from terminal interactive UI is not supported.

@staebler please confirm if `terminal interactive UI` is suitable here, thanks.

**Comment by mjpytlak** [comment]
> @ctauchen @stevsmit [Bug 2051334](https://bugzilla.redhat.com/show_bug.cgi?id=2051334) - [AWS GovCloud] Cluster can not be installed on AWS GovCloud regions via terminal interactive UI. will not be fixed in 4.10 GA, please add it to the release note. thanks
> 
> sample description: Installing cluster into GovCloud regions from terminal interactive UI is not supported.
> 
> @staebler please confirm if `terminal interactive UI` is suitable here, thanks.

@yunjiang29 All support versions of the OCP doc, as well as the 4.10 doc, state that you must manually create/configure the `install-config.yaml` file to deploy a cluster. I believe you are referring using the installer

**Comment by staebler** [comment]
> @ctauchen @stevsmit [Bug 2051333](https://bugzilla.redhat.com/show_bug.cgi?id=2051333) - [aws] records in public hosted zone and BYO private hosted zone were not deleted. will not be fixed in 4.10 GA, please add it to the release note. thanks
> 
> sample description: If you defined pre-existing Route53 private hosted zone when you deploy the cluster, the records in Route53 private and public hosted zone cannot be deleted when destroying the cluster.
> 
> cc @staebler

I have not been able to replicate this issue.

**Comment by bscott-rh** [comment]
> > @ctauchen @stevsmit [Bug 2051333](https://bugzilla.redhat.com/show_bug.cgi?id=2051333) - [aws] records in public hosted zone and BYO private hosted zone were not deleted. will not be fixed in 4.10 GA, please add it to the release note. thanks
> > sample description: If you defined pre-existing Route53 private hosted zone when you deploy the cluster, the records in Route53 private and public hosted zone cannot be deleted when destroying the cluster.
> > cc @staebler
> 
> I have not been able to replicate this issue.

@yunjiang29 In case we do need to add it to the release notes, how does this look:

There is a known issue with this release relating to AWS Route53 DNS records. If you define a private hosted Route53 zone when deploying a cluster, the Route53 records in public and private hosted zones cannot be deleted when destroying the cluster. 

I will hold off on adding this to the release notes until the known issue is fully verified.

**Comment by yunjiang29** [comment]
> @ctauchen @stevsmit [Bug 2051334](https://bugzilla.redhat.com/show_bug.cgi?id=2051334) - [AWS GovCloud] Cluster can not be installed on AWS GovCloud regions via terminal interactive UI. will not be fixed in 4.10 GA, please add it to the release note. thanks
> 
> sample description: Installing cluster into GovCloud regions from terminal interactive UI is not supported.
> 
> @staebler please confirm if terminal interactive UI is suitable here, thanks.
> 
> @yunjiang29 All supported versions of the OCP GovCloud doc, as well as the 4.10 doc, state that you must manually create/configure the install-config.yaml file to deploy a cluster. I believe adding a release note stating that installation program (terminal interactive UI) cannot be used might confuse matters for our end users. While the bug you reference is something that we might want to track for a possible enhancement, I do not believe a release note is required.

@mjpytlak, you are right, user should create install-config.yaml manually, release note is not required for this issue.

**Comment by yunjiang29** [comment]
> > > @ctauchen @stevsmit [Bug 2051333](https://bugzilla.redhat.com/show_bug.cgi?id=2051333) - [aws] records in public hosted zone and BYO private hosted zone were not deleted. will not be fixed in 4.10 GA, please add it to the release note. thanks
> > > sample description: If you defined pre-existing Route53 private hosted zone when you deploy the cluster, the records in Route53 private and public hosted zone cannot be deleted when destroying the cluster.
> > > cc @staebler
> > 
> > 
> > I have not been able to replicate this issue.
> 
> @yunjiang29 In case we do need to add it to the release notes, how does this look:
> 
> There is a known issue with this release relating to AWS Route53 DNS records. If you define a private hosted Route53 zone when deploying a cluster, the Route53 records in public and private hosted zones cannot be deleted when destroying the cluster.
> 
> I will hold off on adding this to the release notes until the known issue is fully verified.

[bscott-rh](https://github.com/bscott-rh) release note is not required for this issue, sorry about the confusing information.

**Comment by rioliu-rh** [comment]
[config drift monitor](https://deploy-preview-42704--osdocs.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-10-release-notes.html#ocp-4-10-mco-config-drift) is not part of machine-api, it is new feature of machine-config-operator
